tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/LockFreeExponentiallyDecayingReservoir.java

```diff
@@ -135,7 +155,11 @@ public void accept(Double priority, WeightedSample sample) {
     }
 
     private LockFreeExponentiallyDecayingReservoir(int size, double alpha, Duration rescaleThreshold, Clock clock) {
-        this.alpha = alpha;
+        // Scale alpha to nanoseconds
+        this.alphaNanos = alpha * SECONDS_PER_NANO;
+        if (Double.compare(alphaNanos, 0) == 0) {
+            throw new IllegalArgumentException("Alpha value " + alpha + " is to small to be scaled to nanoseconds");
```

Comment:
Under which circumstances would we now throw here? Is this a case we need we should test? Should we throw a SafeIllegalArgumentException?

Reply:
I was worried about the Double.MIN_VALUE^2 case where the input is small enough that the conversion updates it to zero.
In retrospect I don't think the validation is meaningful since alpha=0 was allowed for the exponential decay reservoir. In the case alpha is close to zero, decay would take much longer than any jvm will be alive.

Reply:
I'll update the upstream pr as well